gokce_fingerprint_verification.py

```diff
@@ -0,0 +1,184 @@
+######################################################################
+# Names of any others you worked with:
+# AI transcript if used: Claude at moments I got really stuck.
+# Any extensions done:
+# Name: Gokce Can Ertem
+######################################################################
+
+# Function to read the file and turn it into a fingerprint
+def read_and_list_fprint(filename):
+    """Reads the given print and returns a dictionary with all info."""
+    # Your turn!
+    
+
+
+    with open (filename) as fprint:
+        #lines = fprint.read().strip().splitlines()
+        lines = fprint.read().splitlines()
+        fprint2dlist = [list(line) for line in lines]
+        #print("2d List Creating")
+        return(fprint2dlist)
+        #print(fprint2dlist)
+
+
+# Function to find the top of the fingerprint
+def find_row_with_pattern(list_2d, value, count):
+    """
+    Find the first row in a 2D list that contains at least 'count' occurrences of 'value'.
+    
+    Args:
+    list_2d (list): The 2D list to search.
+    value: The value to search for.
+    count (int): The minimum number of occurrences required.
+    
+    Returns:
+    int: The index of the first row that matches the criteria, or -1 if not found.
+    """
+    for i in range(len(list_2d)):
+        if list_2d[i].count(value) >= count:
+            return i
+
+# Function to check original and variant fingerprints
+def simple_and_variant_check(filename1, filename2, pattern_value, pattern_count, rows_to_compare=None):
+    """
+    Compare two 2D lists and return the result.
+    """
+
+
+    #  Get 2d Lists from files
+    list1 = read_and_list_fprint(filename1)
+    list2 = read_and_list_fprint(filename2)
+
+
+
+    # Get the top of the fingerprint
+    start_row1 = find_row_with_pattern(list1, pattern_value, pattern_count)
+    start_row2 = find_row_with_pattern(list2, pattern_value, pattern_count)
+    
+
+    # Find the end of the fingerprint
+    end_row1 = start_row1 + rows_to_compare
+    end_row2 = start_row2 + rows_to_compare
+
+
+
+    total_elements = 0
+    matching_elements = 0
+    #compares the two lists against eachother
+    for row1, row2 in zip(list1[start_row1:end_row1], list2[start_row2:end_row2]):
+        # adds 1 to total elements at every step
+        for item1, item2 in zip(row1, row2):
+            total_elements += 1
+            # adds 1 basen on matching element
+            if item1 == item2:
+                matching_elements += 1 
```

Comment:
This I'm totally fine with, though I find it a bit more transparent here to just loop over shared indexes instead of zipping the lists together, but this works.